applications/nrf_desktop/src/modules/fast_pair.c

```diff
@@ -119,10 +134,38 @@ static bool handle_module_state_event(const struct module_state_event *event)
 	return false;
 }
 
+static bool handle_bt_app_id_type_event(const struct bt_app_id_type_event *event)
+{
+	size_t dongle_peer_cnt = 0;
+
+	for (size_t i = 0; i < ARRAY_SIZE(event->bt_app_id_type); i++) {
+		LOG_DBG("Bluetooth app identity: id %d, type %d", i, event->bt_app_id_type[i]);
+
+		if (event->bt_app_id_type[i] == BT_APP_ID_TYPE_DONGLE) {
+			dongle_app_id = i;
+			dongle_peer_cnt++;
+		}
+	}
+
+	__ASSERT_NO_MSG(dongle_peer_cnt == 1);
+
+	return false;
+}
+
 static bool handle_ble_peer_event(const struct ble_peer_event *event)
 {
-	if (event->state == PEER_STATE_DISCONNECTED) {
-		update_fast_pair_ui_indication();
+	if ((event->state == PEER_STATE_DISCONNECTED) && fp_provider_enabled) {
+		int err;
+		struct bt_conn_info info;
+
+		err = bt_conn_get_info(event->id, &info);
+		if (err) {
+			LOG_ERR("Cannot get conn info (err %d)", err);
+		}
+
+		if (!err) {
+			update_fast_pair_ui_indication(info.id);
+		}
```

Comment:
Is there a reason to use two separate `if`s with complementary checks instead of one `if-else` or `if` with early return?